### PR TITLE
feat(fs): add Arduino ESP LittleFS driver

### PR DESCRIPTION
**Pull request by PierreRambaud**
### Description of the feature or fix

A driver to be able to use Arduino ESP LittleFS file system.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -1066,6 +1066,13 @@ menu "LVGL configuration"
 			default 0
 			depends on LV_USE_FS_LITTLEFS
 
+		config LV_USE_FS_ARDUINO_ESP_LITTLEFS
+			bool "File system on top of littlefs API"
```
Comment (PGNetHun) [inline Kconfig]: To distinguish from the other littlefs driver, please add "Arduino ESP" words to description:
```suggestion
			bool "File system on top of Arduino ESP littlefs API"
```

File: docs/libs/fs.rst [hunk 2]
```diff
@@ -15,6 +15,7 @@ LVG has built in support for:
 - WIN32 (Windows using Win32 API function .e.g ``CreateFileA``, ``ReadFile``)
 - MEMFS (read a file from a memory buffer)
 - LITTLEFS (a little fail-safe filesystem designed for microcontrollers)
+- Arduino LITTLEFS (a little fail-safe filesystem designed for Arduino ESP)
```
Comment (PGNetHun) [inline docs/libs/fs.rst]:
```suggestion
- Arduino ESP LITTLEFS (a little fail-safe filesystem designed for Arduino ESP)
```

File: src/lv_conf_internal.h [hunk 3]
```diff
@@ -2173,6 +2173,24 @@
     #endif
 #endif
 
+/*API for Arduino LittleFs. */
```
Comment (PGNetHun) [inline src/lv_conf_internal.h:2176]:
```suggestion
/*API for Arduino ESP LittleFS. */
```

**Review by PGNetHun — CHANGES_REQUESTED** [review]
Thanks, only some minor (precise) wordings are needed.

**Review by PGNetHun — APPROVED** [review]

File: src/lv_conf_internal.h (hunk 3 above)
Comment (PGNetHun) [reply src/lv_conf_internal.h:2176]: The comment is not corrected yet, but config name reflects that it's for Arduino ESP, so OK.

**Review by kisvegabor — APPROVED** [review]
Wonderful! :tada: 

Thank you! Can you tell what is the plan with the SD library?

**Comment by PierreRambaud** [comment]
I'm gonna make it in a few days, sure it'll be ready for the next week!

**Comment by matixan** [comment]
Hello guys
I'm more than little confused about using this driver. Docs are unclear to me. 
How is `lv_littlefs_set_handler(...)` supposed to work?
What header should I include to use this function, and what is an example parameter?
I want to use this driver with File explorer example for selecting configuration files.
My app is working with SDMMC Arduino library. MCU is ESP32-S3. 
I have access to files in my non GUI app, but file explorer gives me error nr 3 when `show_dir` is called, like it's not correctly initialised.

**Comment by PierreRambaud** [comment]
> lv_littlefs_set_handler

The `lv_littlefs_set_handler` is related to `lfs.h` driver, not the Arduino one. For `lfs.h` you have to follow the instruction provided by the original repository: https://github.com/littlefs-project/littlefs

If you have Arduino.h installed, you can use the lvgl configuration like I did here: https://github.com/PierreRambaud/lvgl-arduino-sd-example/blob/master/lib/lv_conf/lv_conf.h#L631 and that's it, you should be able to use it like this: https://github.com/PierreRambaud/lvgl-arduino-sd-example/blob/master/src/arduino-sd.cpp#L42

**Comment by matixan** [comment]
Sorry for trouble
Now i see that in master branch there are arduino related files.
I'll check them out.

**Comment by PierreRambaud** [comment]
> Sorry for trouble Now i see that in master branch there are arduino related files. I'll check them out.

It will probably be easier, I had a lot of trouble playing with the original lfs.h https://github.com/PierreRambaud/pokegotchi/commit/eec197e5deb2c412d7e18e27ae00a2ab78b050e9 :sweat_smile:

**Pull request merged**